test/functional/wallet_listsinceblock.py

```diff
@@ -167,6 +168,31 @@ def test_reorg(self):
         found = next(tx for tx in transactions if tx['txid'] == senttx)
         assert_equal(found['blockheight'], self.nodes[0].getblockheader(nodes2_first_blockhash)['height'])
 
+    def test_cant_read_block(self):
+        self.log.info('Test the RPC error "Can\'t read block from disk"')
+
+        # Split network into two
+        self.split_network()
+
+        # generate on both sides
+        nodes1_last_blockhash = self.generate(self.nodes[1], 6, sync_fun=lambda: self.sync_all(self.nodes[:2]))[-1]
+        self.generate(self.nodes[2], 7, sync_fun=lambda: self.sync_all(self.nodes[2:]))[0]
+
+        self.join_network()
+
+        # Renaming the block file to induce unsuccessful block read
+        blk_dat = (self.nodes[0].blocks_path / "blk00000.dat")
+        blk_dat_moved = blk_dat.rename(self.nodes[0].blocks_path / "blk00000.dat.moved")
+        assert not blk_dat.exists()
+
+        # listsinceblock(nodes1_last_blockhash) should now fail as blocks are not accessible
+        assert_raises_rpc_error(-32603, "Can't read block from disk",
+            self.nodes[0].listsinceblock, nodes1_last_blockhash)
```

Comment:
I've been trying to understand how does calling `listsinceblock nodes1_last_blockhash` on `node0` is causing the last condition to be true here: https://github.com/bitcoin/bitcoin/blob/master/src/wallet/rpc/transactions.cpp#L663

The comment above this line talks about requesting a reorg'ed block, but isn't the last blockhash on node0 part of the main chain from POV of node0?

Reply:
There is a reorg happening, nodes 0 and 1 mine 6 blocks on top of the common ancestor, while 2 and 3 mine 7 blocks. When the network is rejoined, the 7 block branch reorgs the 6 block branch. However, `nodes1_last_blockhash` is the tip of the 6 block branch which is not in the main chain.

Reply:
Understood now!
Thanks @achow101, this is very helpful.